parsl/executors/high_throughput/zmq_pipes.py

```diff
@@ -23,12 +23,30 @@ def __init__(self, ip_address, port_range):
 
         """
         self.context = zmq.Context()
-        self.zmq_socket = self.context.socket(zmq.REQ)
-        self.port = self.zmq_socket.bind_to_random_port("tcp://{}".format(ip_address),
-                                                        min_port=port_range[0],
-                                                        max_port=port_range[1])
+        self.ip_address = ip_address
+        self.port_range = port_range
+        self.port = None
+        self.port = self.create_socket_and_bind()
```

Comment:
this looks weird - make create_socket_and_bind change self.port itself. maybe also make it create zmq.Context inside there.

Reply:
This also resets port to ‘None’ at the moment as create doesn’t have a return value.

Reply:
Fixed.